src/useAuth.ts

```diff
@@ -1,12 +1,12 @@
 /* eslint @typescript-eslint/explicit-function-return-type: 0 */
 import { useContext } from 'react';
 import { AuthContextProps } from './AuthContextInterface';
-import { AuthContext } from './AuthContext';
+import { AuthContext, defaultAuthContext } from './AuthContext';
 
-export const useAuth = (): AuthContextProps => {
-  const context = useContext<AuthContextProps | undefined>(AuthContext);
+export const useAuth = (ssr?: boolean): AuthContextProps => {
+  const context = useContext<AuthContextProps>(AuthContext);
   
-  if (!context) {
+  if (!ssr && context === defaultAuthContext) {
     throw new Error('AuthProvider context is undefined, please verify you are calling useAuth() as child of a <AuthProvider> component.');
   }
```

Comment:
Maybe you can add some tests to signal what the intentions behind this is?

Reply:
Yep I can do that! The basic idea is that a context needs to exist in either scenario, but we want to stop if we haven't actually initialized a context before hitting this point. So if the context matches the default one made when first loading `oidc-react` then it will reject it like it does today. If you're server-side though, the check is skipped since it doesn't add any value.